ambry-router/src/test/java/com.github.ambry.router/MockServer.java

```diff
@@ -63,6 +64,7 @@
   private boolean getErrorOnDataBlobOnly = false;
   private final ClusterMap clusterMap;
   private final String dataCenter;
+  private final HashMap<RequestOrResponseType, Integer> requestCounts = new HashMap<>();
```

Comment:
should this be threadsafe (i.e. ConcurrentMap and AtomicInteger)? I guess the test where it is used does not include any multi-threading